### ISSUE TITLE
Client does not respect long running operations

### ISSUE THREAD

**Issue by asimarslan** [hazelcast/hazelcast-csharp-client#112]
.Net client has a hard coded 250 sec of invocation timeout. This prevents submitting long running operations such as executeOnKey or complicated predicate queries. 
After the timeout hit, waiting invocation responded by TimeoutException. But the server side continue to execute and respond the actual result. On this case;

```
Received an unhandled exception from a task!: System.InvalidOperationException: An attempt was made to transition a task to a final state when it had already completed. 
at System.Threading.Tasks.Task`1.set_Result(TResult value) 
at Hazelcast.Client.Spi.SettableFuture`1.set_Result(T value) 
at Hazelcast.Client.Spi.ClientInvocationService.HandleResponseMessage(IClientMessage response) 
at Hazelcast.Client.Spi.ClientInvocationService.<>c__DisplayClass4.<HandleClientMessage>b__1() 
at System.Threading.Tasks.Task.InnerInvoke() 
at System.Threading.Tasks.Task.Execute()
```

Java client has no similar timeout and waits the server side forever. The trivial solution is to implement the same behaviour.

**Comment by pveentjer** [comment]
Once the operation starts executing; it will not stop. So there are checks in place that the operation isn't started... but once these hurdles are taken.... the operation keeps running till it complete.

**Comment by pveentjer** [comment]
I see 2 problems in the logging:
1: apparently the task isn't removed after the initial completion was done otherwise a later completing result should not be able to find the task to complete.
2: the completion of the task isn't atomic; if the task is completed and it gets completed again, it should not throw an exception. It should just be ignored.. first come first served.

**Comment by asimarslan** [comment]
fixed by #113
